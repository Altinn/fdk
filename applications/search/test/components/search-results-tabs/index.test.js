import React from 'react';
import { shallow } from 'enzyme';
 import ResultTabs from '../../../src/components/search-results-tabs';

test('should render ResultTabs correctly', () => {
  const defaultProps = {
    location: {
      pathname: "/",
      search: "",
      hash: "",
      key: "ublg0e"
    }
  };
  const wrapper = shallow(<ResultTabs {...defaultProps} />);
  expect(wrapper).toMatchSnapshot();

});





/*
describe('ResultTabs', () => {
  let wrapper;
  const defaultProps = {
    location: {
      pathname: "/",
      search: "",
      hash: "",
      key: "ublg0e"
    }
  };

  it ('should render', () => {
    wrapper = shallow(<ResultTabs {...defaultProps} />);
    expect(wrapper).to.have.length(1);
    expect(wrapper.find('.search-results-tabs')).to.have.length(1);
    expect(wrapper.find(Link)).to.have.length(2);
  });

});
*/

