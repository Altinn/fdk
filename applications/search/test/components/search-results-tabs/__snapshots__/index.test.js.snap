// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ResultTabs correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResultsTabs
    countDatasets={null}
    countTerms={null}
    location={
      Object {
        "hash": "",
        "key": "ublg0e",
        "pathname": "/",
        "search": "",
      }
    }
    selectedLanguageCode={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="col-sm-12 col-md-8 col-md-offset-4"
      >
        <ul
          className="search-results-tabs"
        >
          <li
            className="li-active"
          >
            <Link
              aria-label="Link til side for datasett:"
              replace={false}
              to={
                Object {
                  "pathname": "/",
                  "search": "?",
                }
              }
            >
              Datasett
              <span>
                 (
                )
              </span>
            </Link>
          </li>
          <li
            className=""
          >
            <Link
              aria-label="Link til side for begrep:"
              replace={false}
              to={
                Object {
                  "pathname": "/concepts",
                  "search": "?",
                }
              }
            >
              Begrep
              <span>
                 (
                )
              </span>
            </Link>
          </li>
        </ul>
      </div>,
      "className": "row",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="search-results-tabs"
        >
          <li
            className="li-active"
          >
            <Link
              aria-label="Link til side for datasett:"
              replace={false}
              to={
                Object {
                  "pathname": "/",
                  "search": "?",
                }
              }
            >
              Datasett
              <span>
                 (
                )
              </span>
            </Link>
          </li>
          <li
            className=""
          >
            <Link
              aria-label="Link til side for begrep:"
              replace={false}
              to={
                Object {
                  "pathname": "/concepts",
                  "search": "?",
                }
              }
            >
              Begrep
              <span>
                 (
                )
              </span>
            </Link>
          </li>
        </ul>,
        "className": "col-sm-12 col-md-8 col-md-offset-4",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="li-active"
            >
              <Link
                aria-label="Link til side for datasett:"
                replace={false}
                to={
                  Object {
                    "pathname": "/",
                    "search": "?",
                  }
                }
              >
                Datasett
                <span>
                   (
                  )
                </span>
              </Link>
            </li>,
            <li
              className=""
            >
              <Link
                aria-label="Link til side for begrep:"
                replace={false}
                to={
                  Object {
                    "pathname": "/concepts",
                    "search": "?",
                  }
                }
              >
                Begrep
                <span>
                   (
                  )
                </span>
              </Link>
            </li>,
          ],
          "className": "search-results-tabs",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                aria-label="Link til side for datasett:"
                replace={false}
                to={
                  Object {
                    "pathname": "/",
                    "search": "?",
                  }
                }
              >
                Datasett
                <span>
                   (
                  )
                </span>
              </Link>,
              "className": "li-active",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "aria-label": "Link til side for datasett:",
                "children": Array [
                  "Datasett",
                  <span>
                     (
                    )
                  </span>,
                ],
                "replace": false,
                "to": Object {
                  "pathname": "/",
                  "search": "?",
                },
              },
              "ref": null,
              "rendered": Array [
                "Datasett",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " (",
                      null,
                      ")",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    " (",
                    null,
                    ")",
                  ],
                  "type": "span",
                },
              ],
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                aria-label="Link til side for begrep:"
                replace={false}
                to={
                  Object {
                    "pathname": "/concepts",
                    "search": "?",
                  }
                }
              >
                Begrep
                <span>
                   (
                  )
                </span>
              </Link>,
              "className": "",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "aria-label": "Link til side for begrep:",
                "children": Array [
                  "Begrep",
                  <span>
                     (
                    )
                  </span>,
                ],
                "replace": false,
                "to": Object {
                  "pathname": "/concepts",
                  "search": "?",
                },
              },
              "ref": null,
              "rendered": Array [
                "Begrep",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " (",
                      null,
                      ")",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    " (",
                    null,
                    ")",
                  ],
                  "type": "span",
                },
              ],
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="col-sm-12 col-md-8 col-md-offset-4"
        >
          <ul
            className="search-results-tabs"
          >
            <li
              className="li-active"
            >
              <Link
                aria-label="Link til side for datasett:"
                replace={false}
                to={
                  Object {
                    "pathname": "/",
                    "search": "?",
                  }
                }
              >
                Datasett
                <span>
                   (
                  )
                </span>
              </Link>
            </li>
            <li
              className=""
            >
              <Link
                aria-label="Link til side for begrep:"
                replace={false}
                to={
                  Object {
                    "pathname": "/concepts",
                    "search": "?",
                  }
                }
              >
                Begrep
                <span>
                   (
                  )
                </span>
              </Link>
            </li>
          </ul>
        </div>,
        "className": "row",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ul
            className="search-results-tabs"
          >
            <li
              className="li-active"
            >
              <Link
                aria-label="Link til side for datasett:"
                replace={false}
                to={
                  Object {
                    "pathname": "/",
                    "search": "?",
                  }
                }
              >
                Datasett
                <span>
                   (
                  )
                </span>
              </Link>
            </li>
            <li
              className=""
            >
              <Link
                aria-label="Link til side for begrep:"
                replace={false}
                to={
                  Object {
                    "pathname": "/concepts",
                    "search": "?",
                  }
                }
              >
                Begrep
                <span>
                   (
                  )
                </span>
              </Link>
            </li>
          </ul>,
          "className": "col-sm-12 col-md-8 col-md-offset-4",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="li-active"
              >
                <Link
                  aria-label="Link til side for datasett:"
                  replace={false}
                  to={
                    Object {
                      "pathname": "/",
                      "search": "?",
                    }
                  }
                >
                  Datasett
                  <span>
                     (
                    )
                  </span>
                </Link>
              </li>,
              <li
                className=""
              >
                <Link
                  aria-label="Link til side for begrep:"
                  replace={false}
                  to={
                    Object {
                      "pathname": "/concepts",
                      "search": "?",
                    }
                  }
                >
                  Begrep
                  <span>
                     (
                    )
                  </span>
                </Link>
              </li>,
            ],
            "className": "search-results-tabs",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  aria-label="Link til side for datasett:"
                  replace={false}
                  to={
                    Object {
                      "pathname": "/",
                      "search": "?",
                    }
                  }
                >
                  Datasett
                  <span>
                     (
                    )
                  </span>
                </Link>,
                "className": "li-active",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "aria-label": "Link til side for datasett:",
                  "children": Array [
                    "Datasett",
                    <span>
                       (
                      )
                    </span>,
                  ],
                  "replace": false,
                  "to": Object {
                    "pathname": "/",
                    "search": "?",
                  },
                },
                "ref": null,
                "rendered": Array [
                  "Datasett",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        " (",
                        null,
                        ")",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      " (",
                      null,
                      ")",
                    ],
                    "type": "span",
                  },
                ],
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  aria-label="Link til side for begrep:"
                  replace={false}
                  to={
                    Object {
                      "pathname": "/concepts",
                      "search": "?",
                    }
                  }
                >
                  Begrep
                  <span>
                     (
                    )
                  </span>
                </Link>,
                "className": "",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "aria-label": "Link til side for begrep:",
                  "children": Array [
                    "Begrep",
                    <span>
                       (
                      )
                    </span>,
                  ],
                  "replace": false,
                  "to": Object {
                    "pathname": "/concepts",
                    "search": "?",
                  },
                },
                "ref": null,
                "rendered": Array [
                  "Begrep",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        " (",
                        null,
                        ")",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      " (",
                      null,
                      ")",
                    ],
                    "type": "span",
                  },
                ],
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
